src/providers/address/address.spec.ts

```diff
@@ -57,135 +61,78 @@ describe('AddressProvider', () => {
+      'bitcoincash:qz8ds306px5n65gffn8u69vvnksfw6huwyjczrvkh3?amount=0.00090000'
+    ];
 
-      BTCTestnetAddress = 'bitcoin:n3LHh1WTFSpSVKXNFQo4U5eLAqowCadFHY';
-      result = addressProvider.getNetwork(BTCTestnetAddress);
-      expect(result).toEqual('testnet');
-    });
+    const BCHTestnetAddresses = [
+      // 'mg6PLV5yyUS6Gy55fJ7f994dQ7RpfJNYC9', // TODO: BCH Testnet Bitpay Style
+      'qqr99gt5qdk4qyaxxvzeapgjuxkg6x9ueue95fakj7',
+      'bitcoincash:qqycye950l689c98l7z5j43n4484ssnp4y3uu4ramr',
+      'BCHTEST:qqycye950l689c98l7z5j43n4484ssnp4y3uu4ramr',
+      'BCHTEST:qqycye950l689c98l7z5j43n4484ssnp4y3uu4ramr?amount=0.00090000'
+    ];
 
-    it('should return correct network for BCH livenet address', () => {
-      let BCHLivenetAddress = 'qq9jk8jskjsmgqnzygwjsghp3knautm2dcnc5e4k7n';
-      let result = addressProvider.getNetwork(BCHLivenetAddress);
-      expect(result).toEqual('livenet');
+    it("should get 'livenet' string if address is BTC Livenet", () => {
+      BTCLivenetAddresses.forEach(BTCLivenetAddress => {
+        expect(addressProvider.getNetwork(BTCLivenetAddress)).toEqual(
+          'livenet'
+        );
+      });
+    });
 
-      BCHLivenetAddress =
-        'BITCOINCASH:qz8ds306px5n65gffn8u69vvnksfw6huwyjczrvkh3';
-      result = addressProvider.getNetwork(BCHLivenetAddress);
-      expect(result).toEqual('livenet');
+    it("should get 'testnet' string if address is BTC Testnet", () => {
+      BTCTestnetAddresses.forEach(BTCTestnetAddress => {
+        expect(addressProvider.getNetwork(BTCTestnetAddress)).toEqual(
+          'testnet'
+        );
+      });
     });
 
-    it('should return correct network for BCH testnet address', () => {
-      let BCHTestnetAddress = 'qqr99gt5qdk4qyaxxvzeapgjuxkg6x9ueue95fakj7';
-      let result = addressProvider.getNetwork(BCHTestnetAddress);
-      expect(result).toEqual('testnet');
+    it("should get 'livenet' string if address is BCH Livenet", () => {
+      BCHLivenetAddresses.forEach(BCHLivenetAddress => {
+        expect(addressProvider.getNetwork(BCHLivenetAddress)).toEqual(
+          'livenet'
+        );
+      });
+    });
 
-      BCHTestnetAddress = 'BCHTEST:qqycye950l689c98l7z5j43n4484ssnp4y3uu4ramr';
-      result = addressProvider.getNetwork(BCHTestnetAddress);
-      expect(result).toEqual('testnet');
+    it("should get 'testnet' string if address is BCH Testnet", () => {
+      BCHTestnetAddresses.forEach(BCHTestnetAddress => {
+        expect(addressProvider.getNetwork(BCHTestnetAddress)).toEqual(
+          'testnet'
+        );
+      });
     });
 
-    it('should return undefined if we send an invalid address', () => {
-      let address = 'invalidAddress';
-      let result = addressProvider.getNetwork(address);
+    it('should return null if we send an invalid address', () => {
+      const address = 'invalidAddress';
+      const result = addressProvider.getNetwork(address);
       expect(result).toBeUndefined();
```

Comment:
should return null or undefined?

Reply:
The getNetwork function currently returns undefined. That test works correctly